src/hmem_cuda.c

```diff
@@ -312,18 +480,73 @@ int cuda_hmem_init(void)
 
 	gdrcopy_ret = cuda_gdrcopy_hmem_init();
```

Comment:
Since this sets up a thread and does some other init the param should be checked first so the init can be skipped when gdrcopy is disabled.